public/crclib.c

```diff
@@ -112,65 +112,42 @@ void CRC32_ProcessByte( dword *pulCRC, byte ch )
 void CRC32_ProcessBuffer( dword *pulCRC, const void *pBuffer, int nBuffer )
 {
 	dword	ulCrc = *pulCRC, tmp;
-	byte	*pb = (byte *)pBuffer;
-	uint	nFront;
-	int	nMain;
-JustAfew:
-	switch( nBuffer )
+	dword	*pb = (dword *)pBuffer;
+	byte	*pbc = NULL;
+
+	while(nBuffer >= sizeof(long long))
```

Comment:
Use integer64, C89 does not have long long.
https://github.com/FWGS/xash3d-fwgs/blob/master/common/xash3d_types.h#L32